<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Brian C. O’Meara" />


<title>PhyloMeth: Getting Trees and Data</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-103450-7"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-103450-7');
</script>


<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PhyloMeth</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="syllabus.html">Syllabus</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Exercises
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="gettingstarted.html">Getting Started</a>
    </li>
    <li>
      <a href="gettingtreesanddata.html">Stealing Trees And Data</a>
    </li>
    <li class="dropdown-header">Building Trees</li>
    <li class="dropdown-header">Dating Trees</li>
    <li>
      <a href="discrete.html">Discrete Traits</a>
    </li>
    <li>
      <a href="continuous.html">Continuous Traits</a>
    </li>
    <li>
      <a href="correlations.html">Correlated Traits</a>
    </li>
    <li>
      <a href="diversification.html">Diversification</a>
    </li>
    <li class="dropdown-header">Simulation</li>
  </ul>
</li>
<li>
  <a href="https://www.youtube.com/channel/UCoE9-jWQBwC3iyzMBE3nJNA/">Videos</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Book
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="https://bookdown.org/bomeara/comparative-methods/">HTML version</a>
    </li>
    <li>
      <a href="https://bookdown.org/bomeara/comparative-methods/comparative-methods.pdf">PDF version</a>
    </li>
    <li>
      <a href="https://github.com/bomeara/ComparativeMethodsInR">Git repo</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">PhyloMeth: Getting Trees and Data</h1>
<h4 class="author">Brian C. O’Meara</h4>
<h4 class="date">02 March, 2021</h4>

</div>


<p>Let’s get some trees from Open Tree of Life. This is in some ways a successor to TreeBASE: another repository of trees (though it has only partial overlap with the trees in TreeBASE (or in another repository, Dryad). Unlike TreeBASE, anyone, not just the author, can add trees to Open Tree’s repository. Another important distinction is that Open Tree seeks to create a single tree for all life by creating essentially a supreme super tree.</p>
<p>Now let’s get ya tree, replacing the <code>eval=FALSE</code> with <code>eval=TRUE</code> and replacing the missing info.</p>
<pre class="r"><code>GetTreeFromOpenTree &lt;- function() {
    library(rotl)
    library(ape)

 formica.id &lt;- rotl::tnrs_match_names(&quot;__________&quot;)$ott_id

    # Now get Open Tree&#39;s current best estimate of the phylogeny for the group
    # They call this the tree of life; we can get the subtree for just this group.
    formica.tree &lt;- rotl::tol_subtree(ott_id=formica.id)

    # Let&#39;s plot the tree:
    ape::plot.phylo(&quot;__________&quot;, type=&quot;fan&quot;, cex=0.2)
}</code></pre>
<p>It has a lot of polytomies, representing uncertainty. A maximally resolved tree (if rooted) will have one fewer internal nodes than terminal nodes: think of a tree with three taxa, ((A,B),C): it will have the MRCA of A and B and the MRCA of A, B, and C: three terminals, two internal nodes. If it had no information, it would only have one node. So we can look at the ratio of number of internal nodes to number of possible internal nodes to figure out how resolved a tree is (subtracting 1 from each to account for the root node that must always exist)</p>
<pre class="r"><code>    print(paste(&quot;The formica tree has &quot;, ape::Ntip(formica.tree), &quot; terminals and &quot;,
    Nnode(formica.tree), &quot; internal nodes out of &quot;,ape::Ntip(formica.tree)-2,
    &quot; possible, which means it is &quot;,
    round(100*(ape::Nnode(formica.tree)-1)/(ape::Ntip(formica.tree)-3), 2),
    &quot;% resolved&quot;, sep=&quot;&quot;))

    # Open Tree can also return the original studies with the source trees.
    formica.trees &lt;- studies_find_trees(property=&quot;ot:ottTaxonName&quot;, value=&quot;Formica&quot;, detailed=FALSE)
    formica.studies.ids &lt;- unlist(formica.trees$study_ids)

    # Let&#39;s get info on the first study
    formica.study1.metadata &lt;- rotl::get_study_meta(formica.studies.ids[1])
    print(rotl::get_publication(formica.study1.metadata))

    # And let&#39;s get the tree from this study
  formica.study1.tree1 &lt;- get_study(formica.studies.ids[1])[[1]]

    # And plot it
    ape::plot.phylo(formica.study1.tree1, type=&quot;fan&quot;, cex=0.2)
}</code></pre>
<p>Another question is where to get data. One important way is to collect your own: go out and measure seed size, count insect hairs, measure polar bear weight, etc. However, another way is to gather data already published by others. As with trees, it is important to <em>cite your sources</em>. People have put a lot of work into gathering data, and citation is the main way they get credit. It is also an essential way to reward people who choose to share data for others to build upon, correct, and check for reproducibility (some researchers still choose not to share data). Having citations also lets you and future scientists check for problems.</p>
<p>There are many places to get data. Perhaps the most convenient is to use <a href="https://ropensci.org/packages/">rOpenSci’s packages</a>, which have interfaces to places like <a href="https://github.com/ropensci/bold">Barcode of Life</a>, <a href="https://github.com/ropensci/rgbif">GBIF</a>, <a href="https://github.com/ropensci/reol">Encyclopedia of Life</a>, <a href="https://github.com/ropensci/neotoma">Neotoma paleoecological database</a>, <a href="https://github.com/ropensci/rfishbase">Fishbase</a>, and much more. A major source for datasets and supplemental from particular papers is the <a href="http://datadryad.org">Dryad</a> site. For plant traits, the <a href="https://www.try-db.org/TryWeb/Home.php">TRY database</a>, which has many different datasets, can be useful, though policies on sharing can differ by dataset. <a href="http://www.morphbank.net">Morphbank</a> has biological images while <a href="https://morphobank.org">Morphobank</a> has images and other phenotypic data. Katherine Bannar-Martin has a <a href="http://www.khbannarmartin.com/online-databases/">list</a> of databases for biological anthropology, mammals, fish, other vertebrates, plants, and more.</p>
<p>For homework, consider what biological question you’re curious about and gather data for it. Importantly, <strong>look at the data</strong> before using it. Data are messy. For example, take latitude and longitude. Nice, continuous numbers: not “smooth vs hairy” but something you just read from your GPS – sure, there may be some imprecision, but it’s not that bad, right? However, such data are full of errors: incorrect taxonomy, dropping signs or direction labels (140.5 W longitude is -140.5, not 140.5), entering 0,0 for missing data rather than leaving blank, recording the location of the collection where the specimen is housed rather than where it was collected, recording location of specimens under cultivation or in captivity (a polar bear in the San Diego Zoo), or just simple errors. Other kinds of data have their own problems: fish length – is it snout to end of tail fin, snout to vent? Is it for the biggest adult (beware indeterminate growth), average adult, whatever fish, adult or juvenile, was caught? For plant growth habit, how are woody vines counted: as woody (since they have woody structure) or herbaceous (not self supporting or tall)? Any time continuous variation is put into discrete bins, there are weird corner cases – after all, it’s very rare for a trait to change instantly in one generation, even seemingly discrete ones like presence of eyes (what about cave fish?) or number of limbs (skinks? male boa constrictors with pelvic spurs?).</p>
<pre class="r"><code>    # Get data from an external source
    # Load the data in
    # Plot the data, summarize the data, etc. to make sure there are no weird values.</code></pre>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
