---
title: 'PhyloMeth: Syllabus'
author: "Brian O'Meara"
output:
  pdf_document: default
  html_document: default
  word_document: default
---

## Phylogenetic Methods (PhyloMeth). Spring 2019

UT students may sign up at EEB 587.

Syllabus version: `r format(Sys.time(), '%I:%M %p, %d %B, %Y')`

Instructor: **[Brian O'Meara](http://www.brianomeara.info)** (bomeara@utk.edu)

Course website: http://phylometh.info

Office Hours (446 Hesler) by appointment (given few enrolled students this semester, being flexible with time makes sense).

#### Anonymous feedback

http://www.brianomeara.info/feedback

#### What you will learn
This is a modern class in phylogenetic methods. You will learn how to get data, build a tree, use comparative methods, and how to make new methods. More importantly, you will learn **why** to do all this.

#### Requirements
* A laptop on which you can install software (have root / admin access)
    + It'll be easiest if this is a Mac; less easy is Linux; Windows will be hard
* A question and some data to answer it
* GitHub account
* Slack
* R
* And other software we'll learn about during the course

I am revising a textbook as part of the course
* [HTML version](https://bookdown.org/bomeara/comparative-methods/)
* [PDF](https://bookdown.org/bomeara/comparative-methods/comparative-methods.pdf)

It will be updated as we go, so check back frequently.

#### Evaluation
Grading will be based on effort and performance. Some of the things will be harder for some students than others just given differences in what they've been exposed to so far, and I don't want to penalize students who are less skilled at R, for example. But I do require that you put in work on this class: work through the exercises, think about the papers, etc. Much of the homework is chewy, so it will take some thought to get through, and you might not be able to do all of this. That's ok. *But ask for help!*

These projects should be able to serve as the core for a dissertation chapter or published paper: still requiring more work and writing, but a solid idea with preliminary results.

#### Teaching
The class is generally flipped: rather than me droning at the front of the room, I'll be releasing videos online for you to watch. They will be linked to from this syllabus. In class, talk about what was not clear from the videos about methods, and we'll dig into that. We can also work on problematic parts of the exercises (you should do most of them before class) and talk about papers.

#### Connecting

We will meet via Zoom. The class meets 11:10 am - 12:25 pm US Eastern time. You can download a calendar files of meeting times [here](https://tennessee.zoom.us/meeting/946522241/ics?icsToken=261f8071e9dfcf3384ab995c88f5bfaa8ec029c0e926ca7747c7cbe490136e22) (note I have the times so the meeting starts 40 minutes after class officially starts so you can join early). You can join the meeting at https://tennessee.zoom.us/j/946522241 (it will have you download the free Zoom app). You can call in ([phone numbers for various countries](https://zoom.us/u/adNNl8oCNK); the meeting id is `946 522 241`.

The class discussion board is public on Slack; you will need the [Slack invite code](https://join.slack.com/t/phylometh/shared_invite/enQtNTE5ODEzNDk5MDkyLTg0ZDQ2NTdjMjg1ZWIxZjMzNTg4MDQ0ZmU2MmFiMTY1YTNlZDcyNTg2MjBkZGY4MGQ2MWM0MDkyZGQ2MDRmMWY) to join.

#### Online option
There is a no-credit online option, as well.

### Schedule

* Week 1, Jan 10
    + Topic: Getting ready
    + Items: R, github, gitter, unit testing
    + Instructions & exercise: [Getting Started](gettingstarted.html)
    + One way to do the exercise is to fork the [github repo](https://github.com/bomeara/phylomethinfo), clone it to your computer, start editing documents, then `git commit -m"done homework" -a`, then do a pull request.

* Week 2, Jan 15 & Jan 17
    + Topic: Gathering data
    + Reading: [Chapter 3](https://bookdown.org/bomeara/comparative-methods/getting-data-and-trees-into-r.html) of the textbook.
    + Videos: See videos in this [playlist](https://www.youtube.com/playlist?list=PLrJfyqNDuTkMx_1OV_97S0jgdfmHfQ48x)
    + Items: Sequences, trees, name resolution, alignment
    + Exercise: Getting trees: https://github.com/bomeara/phylometh/blob/master/exercises/gettingtreesanddata/exercise.R
    + In class: Discussion of what kinds of data you need for your questions, what kind of trees you need.

* Week 3, Jan 22 & Jan 24
    + Topic: Building trees
    + Videos: [Terminology](https://www.youtube.com/watch?v=XSp1gEKNuIo), [Tree space](https://www.youtube.com/watch?v=SQw9ZWFp178), [Likelihood vs Bayes](https://youtu.be/xOADfntkBrs)
    + Reading
        + [Felsenstein 1978](https://www.mendeley.com/catalog/cases-parsimony-compatibility-methods-positively-misleading/)
        + [Felsenstein 1985](https://www.mendeley.com/catalog/confidence-limits-phylogenies-approach-using-bootstrap-55/)
        + [Lewis 2001](https://www.mendeley.com/catalog/phylogenetic-systematics-turns-new-leaf/)
        + [Drummond & Rambaut 2007](https://www.mendeley.com/catalog/beast-bayesian-evolutionary-analysis-sampling-trees-1/)
    + Items: Likelihood; Bayes; Parsimony
    + Potential exercises (choose your own)
        + [RAxML tutorial](http://sco.h-its.org/exelixis/web/software/raxml/hands_on.html): Fast maximum likelihood tree inference program.
        + [RevBayes tutorials](https://revbayes.github.io/tutorials.html): Bayesian program with many different models.
        + [BEAST tutorials](http://treethinkers.org/tutorials/divergence-time-estimation-using-beast/): Bayesian tree inference program, especially used for getting chronograms.
        + [Cyverse discovery environment](https://de.cyverse.org/de/): Free service (thanks, NSF!) for running phylogenetic (and other) software


* Week 4, Jan 29 & Jan 31
    + Topic: Building trees
    + Items: Gene tree species tree; Network
    + Reading
        + [Maddison 1997](https://doi.org/10.1093/sysbio/46.3.523): Gene trees in species trees
        + [Heled & Drummond 2009](https://doi.org/10.1093/molbev/msp274): A method for inferring species trees given gene trees inside them
        + [Degnan & Rosenberg 2000](http://dx.doi.org/10.1016/j.tree.2009.01.009): Cases where the most frequent gene tree does not match the species tree
        + [Solís-Lemus & Cécile Ané, 2016](http://dx.doi.org/10.1371/journal.pgen.1005896): Phylogenetic networks. A bit mathy for many students in this class, but worth introducing quartets, reticulation, and an existing method.
        + [Hahn & Nakhleh 2016](http://www.indiana.edu/~hahnlab/Publications/HahnNakhleh2016.pdf): Irrational exuberance for resolved species
trees. Mind-blowing (to me) paper on why to understand trait evolution we may care about the gene trees, not the species tree.
    + Class exercise: [R Markdown](https://raw.githubusercontent.com/bomeara/ComparativeMethodsInR/master/95_GeneTreeSpeciesTree.Rmd)


* Week 5, Feb 5 & Feb 7
    + Topic: Dating trees
    + Items: Beast, r8s
    + Reading
        + [Sanderson, 2002](http://dx.doi.org/10.1093/oxfordjournals.molbev.a003974): Penalized likelihood.
        + [Drummond et al. 2006](http://dx.doi.org/10.1371/journal.pbio.0040088): Bayesian dating. And good paper title.
        + [Drummond and Rambaut, 2007](http://dx.doi.org/10.1186/1471-2148-7-214): BEAST description.
        + [Beaulieu et al., 2015](http://dx.doi.org/10.1093/sysbio/syv027): One possible caveat to Bayesian dating.
        + [Heath et al. 2015](http://www.pnas.org/content/111/29/E2957.short): Using fossils as tips.
        + [Course book](https://bookdown.org/bomeara/comparative-methods/dating.html)
    + Exercise
        + [RevBayes](http://revbayes.github.io/tutorials.html) tutorials on dating.

* Week 6, Feb 12 & Feb 14
    + Topic: Discrete character models
    + Items: corHMM, geiger
    + Video: [PhyloMeth discrete characters](https://youtu.be/0OWX6WMNQ3o?list=PLrJfyqNDuTkOzbZfsM_xIkNIBjF7zmVP4)
    + Reading (all three for Tues)
        + [Pagel, 1999](https://doi.org/10.1080/106351599260184): Ancestral state reconstruction
        + [Lewis, 2001](https://doi.org/10.1080/106351501753462876): MKV model
        + [O'Meara, 2012](https://doi.org/10.1146/annurev-ecolsys-110411-160331): Review of models
    + Exercise (for Thurs): https://github.com/bomeara/phylometh_discrete


* Week 7, Feb 19 & Feb 21
    + Topic: Continuous character models
    + Items: Surface, OUwie, geiger
    + Videos: [OU, BM, etc](https://youtu.be/awN_VBv8reU?list=PLRyq_4VPZ9g9-pxqGRLJjE1FcqNvnC-gg)
    + Reading:
        + [Hansen and Martins, 2006](http://www.jstor.org/stable/2410878): Linking micro and macroevolutionary models. TL;DR: Table 1 shows that many microevolutionary models reduce to Brownian motion.
        + [Felsenstein 1988](http://www.annualreviews.org/doi/abs/10.1146/annurev.es.19.110188.002305?journalCode=ecolsys.1): Biology behind the model.
        + [O'Meara, 2012](https://doi.org/10.1146/annurev-ecolsys-110411-160331): Not nearly as important as the ones above, but an intro to models that shows how I think.
        + [Course Book](https://bookdown.org/bomeara/comparative-methods/brownian-motion-and-correlations.html)
          + Exercise (for Thurs): https://github.com/bomeara/phylometh_continuous


* Week 8, Feb 26 & Feb 28
    + Topic: Correlating characters
    + Items: independent contrasts, Pagel 1994, Hansen OU
    + Reading
        + [Felsenstein, 1985](http://dx.doi.org/10.1086/284325): Independent contrasts
        + [Pagel, 1994](http://dx.doi.org/10.1098/rspb.1994.0006): Pagel 94 correlation
        + [Maddison & FitzJohn, 2015](https://academic.oup.com/sysbio/article-lookup/doi/10.1093/sysbio/syu070): A big problem with correlation (and other issues)
    + Exercise (for Thurs): https://github.com/bomeara/phylometh_correlated


* Week 9, Mar 5 & Mar 7
    + Topic: Biogeography and optimization
    + Reading
      + [Ronquist and Sanmartin, 2011](http://www.annualreviews.org/doi/abs/10.1146/annurev-ecolsys-102209-144710)
      + [Ree and Smith, 2008](https://academic.oup.com/sysbio/article/57/1/4/1703014/Maximum-Likelihood-Inference-of-Geographic-Range)
    + Exercise: None this week, but continue working on past ones.




* Week 10, Mar 12 & Mar 14
    + Topic: Diversification and SSE models
    + Items: diversitree, hisse
        + [Magallon and Sanderson, 2001](http://dx.doi.org/10.1554/0014-3820(2001)055[1762:ADRIAC]2.0.CO;2): As an empirical paper of ages, later work has improved on this, but it has extremely clear explanations of the math behind these methods.
        + [Maddison, FitzJohn, and Otto, 2007](https://doi.org/10.1080/10635150701607033): The BiSSE paper. Download it as a PDF: the equations do not render in the HTML.
        + [Maddison & FitzJohn, 2015](https://academic.oup.com/sysbio/article-lookup/doi/10.1093/sysbio/syu070): Already read it, but reread for this week.
        + [Beaulieu & O'Meara, 2016](https://doi.org/10.1093/sysbio/syw022): A hidden state version of the BiSSE model.
    + Exercise: https://github.com/bomeara/phylometh_diversification

* Spring break at UT March 18-22

* Week 11, Mar 26 & Mar 28
    + Topic: Simulating data
    + Items: TreeSim, geiger

* Week 12, Apr 2 & Apr 4
    + Topic: Building a new comparative method

* Week 13, Apr 9 & Apr 11
    + Topic: Testing a new comparative method

* Week 14, Apr 16 & Apr 18
    + Topic: Student choice
    + Exercise: make an R package based on the NSF API.

* Week 11, Aprl 23 & April 25
    + Topic: Student choice
    + Exercise: Mixed data types. https://github.com/bomeara/phylometh_mixed.

**Course made possible by**:  NSF CAREER grant to O'Meara & ongoing support from the Department of Ecology & Evolutionary Biology, U. of Tennessee, Knoxville.

My goal is to have you learn. If you are having trouble with something in the course, or if there is a topic you just have to learn more about, let me know (email, office hours, online forum, etc.). Faculty often use evaluations at the end of the semester to get info from students so we can improve before the next class, but this does not help you directly. To allow the class to improve while you are taking it, I have created a site for anonymous feedback at http://brianomeara.info/teaching/feedback/ (and yes, it is really anonymous). Let me know things that are going well or poorly — both are important. I might not implement all your suggestions, but they will all be read and considered, generally the same day you submit them.

Any student who feels s/he may need an accommodation based on the impact of a disability should contact me privately to discuss specific needs.  Please contact the Office of Disability Services at 865-974-6087 in Dunford Hall to coordinate reasonable accommodations for students with documented disabilities.

All relevant University policies (including, but not limited to, policies on academic integrity, attendance, etc.) apply to this course. In the case of any conflict between the policies in this syllabus and University policy, University policy applies. The instructor reserves the right to revise, alter, and/or amend this syllabus as necessary. Students will be notified by email of any such revisions, alterations, and/or amendments.
